lib/galaxy/managers/datasets.py

```diff
@@ -341,7 +341,7 @@ def stop_creating_job(self, dataset_assoc):
             job = dataset_assoc.creating_job_associations[0].job
             if not job.finished:
                 # Are *all* of the job's other output datasets deleted?
-                if job.check_if_output_datasets_deleted():
+                if job.check_if_output_datasets_deleted() or self.app.config.delete_output_dataset_kills_job:
```

Comment:
Can you adjust the comment in line 343 and the docstring of this function to reflect the new behavior ?